### PR TITLE
Sanitize user registration payload for non-administrators (#6069)

### PR DESCRIPTION
**Pull request by diogotcorreia**
#### Description of what you did:
This is a fix for #6069.

I've basically added a check for admin users. If it's not an admin, it strips away everything from the payload except for the required fields.

Please tell me if something needs to be changed.

### CODE REVIEW

File: packages/strapi-plugin-users-permissions/controllers/Auth.js [hunk 1]
```diff
@@ -394,10 +394,14 @@ module.exports = {
       );
     }
 
-    const params = _.assign(ctx.request.body, {
+    let params = _.assign(ctx.request.body, {
       provider: 'local',
     });
 
+    if (!ctx.state.admin) {
```
Comment (alexandrebodin) [inline packages/strapi-plugin-users-permissions/controllers/Auth.js:401]: Can you change this to 

```js
const params = Object.assign(_.pick(ctx.request.body, ['username', 'email', 'password']), {
provider: 'local'
})
```

I don't think there is any reason this route should allow other params if it is called by and admin. There are administration routes for this instead

**Comment by alexandrebodin** [comment]
@diogotcorreia As this is a pretty urgent matter I'm going to make the fix directly if you don't mind ;)

**Comment by diogotcorreia** [comment]
> @diogotcorreia As this is a pretty urgent matter I'm going to make the fix directly if you don't mind ;)

I've just made the change, don't worry

**Comment by alexandrebodin** [comment]
Your PR is not passing the checks and I need to make it directly on the exact tag ;) Will make it and set you as coauthor

**Pull request closed without merging**